src/Engine/Network/UDPClient.cpp

```diff
@@ -95,8 +173,55 @@ void UDPClient::Broadcast(Packet& packet, int port)
 
 bool UDPClient::IsSocketAvailable()
 {
-    if (!m_Socket) { 
+    if (!m_Socket) {
         return false;
     }
     return m_Socket->available();
-}
\ No newline at end of file
+}
+
+bool UDPClient::GetNextPacket(Packet & packet)
+{
+    // A duplicate packet should not be present in the vector!
+    // Soo we will assume that this is true and only look if size
+    // of vector is correct.
+    std::map<unsigned int, std::vector<std::pair<int, boost::shared_ptr<char>>>>::iterator it = m_PacketSegmentMap.begin();
+    while (it != m_PacketSegmentMap.end()) {
+        // pair(Group index, packetData)
+        std::vector<std::pair<int, boost::shared_ptr<char>>>& currentVector = it->second;
+        Packet headerInfoPacket(currentVector.at(0).second.get(), packet.HeaderSize());
+        int groupSize = headerInfoPacket.GroupSize();
+        //LOG_INFO("UDPClient::GetNextPacket: Packet group : %i.Group index : %i.Group size : %i. lastReceivedSnapshotGroup: %i. MessageType(Ples 4): %i", headerInfoPacket.Group(), headerInfoPacket.GroupIndex(), groupSize, lastReceivedSnapshotGroup, headerInfoPacket.GetMessageType());
+        //LOG_INFO("UDPClient::GetNextPacket: Packet group : %i.lastReceivedSnapshotGroup: %i. MessageType(Ples 4): %i", headerInfoPacket.Group(), lastReceivedSnapshotGroup, headerInfoPacket.GetMessageType());
+        int mapSize = m_PacketSegmentMap.size();
+        if (mapSize > 5) {
+            it = m_PacketSegmentMap.erase(it);
+            LOG_INFO("The map is increasing in size, size is %i", mapSize);
+            continue;
+        }
+        if (headerInfoPacket.GetMessageType() == MessageType::Snapshot && m_LastReceivedSnapshotGroup > headerInfoPacket.Group()) {
+            it = m_PacketSegmentMap.erase(it);
+            continue;
+            //LOG_INFO("Deleted old entry");
+        }
+        if (currentVector.size() == groupSize) {
+            std::sort(currentVector.begin(), currentVector.end());
+            // Add the first packet in vector
+            packet.ReconstructFromData(currentVector.at(0).second.get(), packet.HeaderSize());
+            // Add the rest of the packets.
+            int sizeOfData = 0;
+            for (size_t i = 0; i < currentVector.size(); i++) {
```

Comment:
Here we could do a `for (auto& x : currentVector)` and swap `currentVector.at(i)` for `x`.